testng-eclipse-plugin/src/main/org/testng/eclipse/util/CustomSuite.java

```diff
@@ -106,7 +109,17 @@ protected XMLStringBuffer createContentBuffer() {
     suiteBuffer.setDocType("suite SYSTEM \"" + Parser.TESTNG_DTD_URL + "\"");
 
     if (hasEclipseXmlFile) {
-      createXmlFileFromTemplate(suiteBuffer, xmlFile);
+      try {
+        IStringVariableManager manager = VariablesPlugin.getDefault().getStringVariableManager();
+        String resolvedXmlFile = manager.performStringSubstitution(xmlFile);

```

Comment:
the absolute path is still supported here, hence no impact on people who already configured with external file (aka, out of workspace)